machine/v1beta1/types_alibabaprovider.go

```diff
@@ -0,0 +1,398 @@
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AlibabaCloudMachineProviderConfig is the Schema for the alibabacloudmachineproviderconfig API
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+// +k8s:openapi-gen=true
+type AlibabaCloudMachineProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// More detail about alibabacloud ECS
+	// https://www.alibabacloud.com/help/doc-detail/25499.htm?spm=a2c63.l28256.b99.727.496d7453jF7Moz
+
+	//The instance type of the instance.
+	InstanceType string `json:"instanceType"`
+
+	// The ID of the vpc
+	// +optional
+	VpcID string `json:"vpcId,omitempty"`
+
+	// The ID of the region in which to create the instance. You can call the DescribeRegions operation to query the most recent region list.
+	RegionID string `json:"regionId"`
+
+	// The ID of the zone in which to create the instance. You can call the DescribeZones operation to query the most recent region list.
+	ZoneID string `json:"zoneId"`
+
+	// The ID of the image used to create the instance.
+	ImageID string `json:"imageId"`
+
+	// DataDisk hold information regarding the extra data disks
+	// +optional
+	DataDisks []DataDiskProperties `json:"dataDisk,omitempty"`
+
+	// SecurityGroups is an array of references to security groups which to assign the instance. The valid values of N vary based on the
+	// maximum number of security groups to which an instance can belong. For more information, see the "Security group limits" section in Limits.
+	// https://www.alibabacloud.com/help/doc-detail/101348.htm?spm=a2c63.p38356.879954.48.78f0199aX3dfIE
+	SecurityGroups []AlibabaResourceReference `json:"securityGroups,omitempty"`
+
+	// Bandwidth describes the internet bandwidth strategy for the instance
+	// +optional
+	Bandwidth BandwidthProperties `json:"bandwidth,omitempty"`
+
+	// Information regarding the the system disk for the instance
+	// +optional
+	SystemDisk SystemDiskProperties `json:"systemDisk,omitempty"`
+
+	// VSwitch is a reference to the vswitch to use for this instance
+	// This parameter is required when you create an instance of the VPC type.
+	// You can call the DescribeVSwitches operation to query the created vSwitches.
+	VSwitch AlibabaResourceReference `json:"vSwitch,omitempty"`
+
+	// Subscription holds the billing information for the instance
+	// +optional
+	Subscription SubscriptionInfo `json:"subscription,omitempty"`
+
+	// Description of the instance. The description must be 2 to 256 characters in length and cannot start with http:// or https://.
+	// This parameter is empty by default.
+	// +optional
+	Description string `json:"description,omitempty"`
```

Comment:
Should we drop this? If this is optional, I guess it's not needed to be set, seems like a feature we don't need to support initially